optimade/validator/validator.py

```diff
@@ -1338,11 +1338,10 @@ def _test_page_limit(
                     f"Error when testing pagination: the response from `links->next` {next_link!r} failed the previous test."
                 )
 
-            check_next_link = bool(check_next_link - 1)
+            check_next_link = check_next_link - 1
             self._test_page_limit(
                 next_response,
                 check_next_link=check_next_link,
-                multistage=check_next_link,
                 previous_links=previous_links,
```

Comment:
Multistage is used in the decorated function (`@test_case`) to prevent validator returning an error for each individual step of a recursive function, so I would leave that in.

Reply:
I'll also investigate a little why this is causing test failures just for the docker image atm

Reply:
I have processed your remarks and I think it should be good to merge now.